### PR TITLE
[API-7] Implement methods for RailDirection and LogAxis

### PR DESCRIPTION
**Pull request by Lignium**
[SpongeAPI](https://github.com/SpongePowered/SpongeAPI/pull/2078) | **SpongeCommon**

Implements the methods that introduced by https://github.com/SpongePowered/SpongeAPI/pull/2078.

### CODE REVIEW

File: src/main/java/org/spongepowered/common/mixin/api/mcp/block/BlockRailBase_EnumRailDirectionMixin_API.java [hunk 1]
```diff
@@ -48,6 +53,78 @@
...
+            case 9:
+                return Direction.NORTH;
+            case 1:
+            case 3:
+                return Direction.EAST;
+            case 4:
+            case 6:
+            case 7:
+                return Direction.SOUTH;
+            case 2:
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/mixin/api/mcp/block/BlockRailBase_EnumRailDirectionMixin_API.java]: All of these magic numbers need to be Constants in the `org.spongepowered.common.util.Constants`, whether it's in a new nested class for Rail Directions or whatever.

File: src/main/java/org/spongepowered/common/mixin/api/mcp/block/BlockLog_EnumAxisMixin_API.java [hunk 2]
```diff
@@ -47,6 +50,20 @@
         return shadow$getName();
     }
 
+    @Override
+    public Optional<Axis> toAxis() {
+        switch ((BlockLog.EnumAxis) (Object) this) {
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/mixin/api/mcp/block/BlockLog_EnumAxisMixin_API.java]: Would it make better sense to just stick the `o.s.a.Axis` as a bridge method on `EnumAxis`? I don't like switch statements as is.

**Review by gabizou — CHANGES_REQUESTED** [review]

File: src/main/java/org/spongepowered/common/mixin/api/mcp/block/BlockLog_EnumAxisMixin_API.java (hunk 2 above)
Comment (Lignium) [reply src/main/java/org/spongepowered/common/mixin/api/mcp/block/BlockLog_EnumAxisMixin_API.java]: @gabizou Еxcuse me, I didn't catch your thought. What is the `o.s.a.`? How can I avoid using `switch` other than replacing them with `if`?

Comment (dualspiral) [reply src/main/java/org/spongepowered/common/mixin/api/mcp/block/BlockLog_EnumAxisMixin_API.java]: `org.spongepowered.api`

**Pull request closed without merging**